### PR TITLE
Remove elasticsearch.rest-test gradle plugin

### PR DESCRIPTION
**Pull request by breskeby**
Removes and remaining usages of `elasticsearch.rest-test` and the plugin itself from the codebase

Relates to https://github.com/elastic/elasticsearch/issues/63696

### CODE REVIEW

File: modules/ingest-geoip/qa/file-based-update/src/test/java/org/elasticsearch/ingest/geoip/UpdateDatabasesIT.java [hunk 1]
```diff
@@ -1,89 +0,0 @@
...
-import java.util.List;
-import java.util.Map;
-
-import static org.hamcrest.Matchers.either;
-import static org.hamcrest.Matchers.equalTo;
-import static org.hamcrest.Matchers.is;
-import static org.hamcrest.Matchers.notNullValue;
-import static org.hamcrest.Matchers.nullValue;
-
-public class UpdateDatabasesIT extends ESRestTestCase {
```
Comment (pugnascotia) [inline modules/ingest-geoip/qa/file-based-update/src/test/java/org/elasticsearch/ingest/geoip/UpdateDatabasesIT.java:30]: Why is this file deleted?

Comment (mark-vieira) [reply modules/ingest-geoip/qa/file-based-update/src/test/java/org/elasticsearch/ingest/geoip/UpdateDatabasesIT.java:30]: Same with the database file itself. Perhaps we inteded to _move_ these?

**Review by mark-vieira — APPROVED** [review]
I think that geoip test got mistakenly removed, otherwise LGTM.

**Comment by breskeby** [comment]
I accidentally have moved the files in an earlier commit but didn't delete the old location. I'll fix this up

File: modules/ingest-geoip/qa/file-based-update/src/test/java/org/elasticsearch/ingest/geoip/UpdateDatabasesIT.java (hunk 1 above)
Comment (breskeby) [reply modules/ingest-geoip/qa/file-based-update/src/test/java/org/elasticsearch/ingest/geoip/UpdateDatabasesIT.java:30]: they have been moved in an earlier pr 🤦 so this is basically a cleanup
